plugins/native/module/vtkF3DSPZReader.cxx

```diff
@@ -0,0 +1,226 @@
+#include "vtkF3DSPZReader.h"
+
+#include <vtkFloatArray.h>
+#include <vtkInformation.h>
+#include <vtkInformationVector.h>
+#include <vtkObjectFactory.h>
+#include <vtkPointData.h>
+#include <vtkPoints.h>
+#include <vtkStreamingDemandDrivenPipeline.h>
+
+#include <vtk_zlib.h>
+
+#include <algorithm>
+
+namespace
+{
+bool UncompressGzip(
+  const std::vector<unsigned char>& compressed, std::vector<unsigned char>& uncompressed)
+{
+  std::vector<uint8_t> buffer(8192);
+  z_stream stream = {};
+  stream.next_in = const_cast<Bytef*>(compressed.data());
+  stream.avail_in = compressed.size();
+  if (inflateInit2(&stream, 16 | MAX_WBITS) != Z_OK)
+  {
+    return false;
+  }
+  bool success = false;
+  while (true)
+  {
+    stream.next_out = buffer.data();
+    stream.avail_out = buffer.size();
+    int32_t res = inflate(&stream, Z_NO_FLUSH);
+    if (res != Z_OK && res != Z_STREAM_END)
+    {
+      break;
+    }
+    uncompressed.insert(
+      uncompressed.end(), buffer.data(), buffer.data() + buffer.size() - stream.avail_out);
+    if (res == Z_STREAM_END)
+    {
+      success = true;
+      break;
+    }
+  }
+  inflateEnd(&stream);
+  return success;
+}
+}
+
+vtkStandardNewMacro(vtkF3DSPZReader);
+
+//----------------------------------------------------------------------------
+vtkF3DSPZReader::vtkF3DSPZReader()
+{
+  this->SetNumberOfInputPorts(0);
+}
+
+//----------------------------------------------------------------------------
+int vtkF3DSPZReader::RequestData(
+  vtkInformation*, vtkInformationVector**, vtkInformationVector* outputVector)
+{
+  vtkPolyData* output = vtkPolyData::GetData(outputVector);
+
+  std::ifstream file(this->FileName, std::ios::binary);
```

Comment:
can you consider using vtkResourceParser/vtkResourceStream instead ?  We will have to transition at some point.

Reply:
good point!